src/Engine.cpp

```diff
@@ -1140,132 +1152,110 @@ bool Engine::update_problem(
+            emit entity_status_proxy_model_->layoutAboutToBeChanged();
 
             emit controller_->update_status_counters(
                     QString::number(controller_->status_counters.total_errors),
                     QString::number(controller_->status_counters.total_warnings));
 
             // remove empty message if exists
-            if (problem_model_->is_empty())
+            if (entity_status_model_->is_empty())
             {
-                problem_model_->removeEmptyItem();
+                entity_status_model_->removeEmptyItem();
             }
 
             // update view
-            problem_proxy_model_->set_source_model(problem_model_);
+            entity_status_proxy_model_->set_source_model(entity_status_model_);
 
-            // notify problem model layout changed to refresh layout view
-            emit problem_proxy_model_->layoutChanged();
+            // notify status model layout changed to refresh layout view
+            emit entity_status_proxy_model_->layoutChanged();
         }
     }
     return true;
 }
 
-bool Engine::update_problem_entities(
+bool Engine::remove_inactive_entities_from_status_model(
         const backend::EntityId& id)
 {
-    // check if there are entities in the problem model
-    if (!problem_model_->is_empty())
+    // check if there are entities in the status model
+    if (!entity_status_model_->is_empty())
     {
         // get info from id
         EntityInfo entity_info = backend_connection_.get_info(id);
 
-        // update problem model if not alive
+        // update status model if not alive
         if (!entity_info["alive"])
         {
             // remove item from tree
-            problem_model_->removeItem(problem_model_->getTopLevelItem(id, "", false, ""));
+            entity_status_model_->removeItem(entity_status_model_->getTopLevelItem(id, "", false, ""));
 
             // add empty item if removed last item
-            if (problem_model_->rowCount(problem_model_->rootIndex()) == 0)
+            if (entity_status_model_->rowCount(entity_status_model_->rootIndex()) == 0)
             {
-                problem_model_->addTopLevelItem(new models::ProblemTreeItem(backend::ID_ALL, std::string("No issues found"), false, std::string("")));
+                entity_status_model_->addTopLevelItem(new models::StatusTreeItem(backend::ID_ALL, std::string("No issues found"), false, std::string("")));
             }
 
-
-            uint32_t error_checker = 0 - 100;
             // update error counter
             std::map<backend::EntityId,uint32_t>::iterator err_it = controller_->status_counters.errors.find(id);
             if(err_it != controller_->status_counters.errors.end())
             {
                 //element found;
                 controller_->status_counters.total_errors -= err_it->second;
-                if (controller_->status_counters.total_errors > error_checker)
+                if (controller_->status_counters.total_errors < 0)
```

Comment:
Same case as in https://github.com/eProsima/Fast-DDS-monitor/pull/204#discussion_r1392072782